index.bs

```diff
@@ -119,105 +133,462 @@ A device is [=capable of supporting=] the depth sensing feature if the device ex
-    1. Initialize |result|'s {{XRDepthInformation/data}} to the raw depth buffer returned in |nativeDepthInformation|.
+    1. Initialize |result|'s {{XRCPUDepthInformation/data}} to the raw depth buffer returned in |nativeDepthInformation|.
     1. Initialize |result|'s [=XRDepthInformation/view=] to |view|.
     1. Return |result|.
 
 </div>
 
-<div class="algorithm" data-algorithm="obtain-depth-at-coordinates">
+<div class="example">
+The following code demonstrates how depth data can be obtained within {{XRFrameRequestCallback}}. It is assumed that the session that has depth sensing enabled, with usage set to "cpu-optimized" and data format set to "luminance-alpha":
 
-When {{XRDepthInformation/getDepth(column, row)}} method is invoked on an {{XRDepthInformation}} |depthInformation| with |column|, |row|, the user agent MUST <dfn>obtain depth at coordinates</dfn> by running the following steps:
-    1. If |column| is greater than or equal to |depthInformation|'s {{XRDepthInformation/width}}, [=exception/throw=] {{NotAllowedError}} and abort these steps.
-    1. If |row| is greater than or equal to |depthInformation|'s {{XRDepthInformation/height}}, [=exception/throw=] {{NotAllowedError}} and abort these steps.
-    1. Let |index| be equal to |row| multiplied by {{XRDepthInformation/width}} & added to |column|.
-    1. Let |depthInMillimeters| be equal to a value found at index |index| in {{XRDepthInformation/data}}.
-    1. Return |depthInMillimeters| divided by <code>1000.0</code>.
+<pre highlight="js">
+
+const session = ...;          // Session created with depth sensing enabled.
+const referenceSpace = ...;   // Reference space created from the session.
+
+function requestAnimationFrameCallback(t, frame) {
+  session.requestAnimationFrame(requestAnimationFrameCallback);
+
+  const pose = frame.getViewerPose(referenceSpace);
+  if(pose) {
+    for(const view of pose.views) {
+      const depthInformation = frame.getDepthInformation(view);
+      if(depthInformation) {
+        useCpuDepthInformation(view, depthInformation);
+      }
+    }
+  }
+}
+</pre>
+
+Once the {{XRCPUDepthInformation}} is obtained, it can be used to discover a distance from the view plane to user's environment (see [[#results-interpretation]] section for details). The below code demonstrates obtaining the depth at normalized view coordinates of (0.25, 0.75):
+
+<pre highlight="js">
+
+function useCpuDepthInformation(view, depthInformation) {
+  // Let's obtain the depth value at normalized view coordinates (0.25, 0.75):
+  const normalizedViewCoordinates = [0.25, 0.75];
+  const normalizedDepthCoordinates = multiplyMatrixByVector(
+    depthInformation.normTextureFromNormView,
+    normalizedViewCoordinates);
+  const depthCoordinates = componentWiseMultipyVectors(
+    normalizedDepthCoordinates,
+    [depthInformation.width, depthInformation.height]);
+  const depthInMeters = depthInformation.getDepthInMeters(
```

Comment:
I know this is pretty late for this feedback, and I'm sorry I missed it before, but I'm mildly surprised to realize that the `getDepthInMeters()` utility function requires pre-transformed coordinates. I was under the impression that doing the texture->view transform was one of the things it would do under the hood to make things easier for users, and if the only thing it's doing is the coordinate lookup and multiply by rawValueToMeters that seems less valuable.

I don't want to introduce a breaking change at this stage, so don't consider this a blocker. Just wanted to voice that this was a disconnect with my expectations.

Reply:
We could make the helper do this work assuming that we change the parameters to take normalized view coordinates & basically perform the same computation that is happening here, so if that is what you have in mind, I'd be OK to make this change. It'd definitely be more ergonomic.

Reply:
Making sure I recall this correctly: Normalized View Coordinates is [0, 0] In the upper left and [1, 1] in the lower right?

Regardless, that does seem like a good ergonomics change if it's not going to be too disruptive. It's far easier to divide by the window size than transform a vector by a matrix.

Reply:
> Normalized View Coordinates is [0, 0] In the upper left and [1, 1] in the lower right?

That's how I have been using them, although the name does not seem to be something that's widely used so I will introduce new item in the terminology and perform those computations in the helper - it will definitely simplify the application code.

Reply:
Resolving, please reopen if the latest changes are insufficient or otherwise lacking.